### DISCUSSION TITLE
What to do with strangers (unknown)?

### DISCUSSION THREAD

**Discussion by miczlo** [exadel-inc/CompreFace#779] [Q&A]
So I've been playing with compreface and while it's quite good when it comes to distinguishing who it spotted from the trained people, when it sees a face that it wasn't trained on it still tries to match it to the known faces and usually gives quite a high result.  

Now how can I prevent this. My logic tells me I shouldn't really be trying to add these strangers to the same class but how else do I prevent compreface from trying to match them to one of the known faces so badly. Ideally I'd like to see a low similarity score to reject it and not something in the 90% range.

**Comment by pospielov** [comment]
Here is documentation that can help with choosing the threshold:
https://github.com/exadel-inc/CompreFace/blob/master/docs/Face-Recognition-Similarity-Threshold.md
In default CompreFace build false positive shouldn't happen so often (just look at the picture in doc), however, as users report, the default build is not so reliable when the quality of the picture is low or with some ethnicities.
A standard recommendation is to use a more modern [CompreFace build](https://github.com/exadel-inc/CompreFace/tree/master/custom-builds), e.g. mobilenet, and improve the picture quality (light, scale, camera location)
